moirae/models/ecm/utils.py

```diff
@@ -58,7 +58,7 @@ def get_value(self, soc: Union[Number, List, np.ndarray]) -> np.ndarray:
 
         # Special case: no interpolation
         if self.base_values.shape[-1] == 1:
-            y = self.base_values[:, 0]
+            y = self.base_values[:, 0].copy()
```

Comment:
Why the copy here? It doesn't look any any of the subsequent actions alter values of y.

Reply:
The subsequent actions don't, but what we do with the values after does. The problem is that `np.reshape` does not return a copy of the array, but a different view of it. For example:
```python
a = np.arange(10)
print(a)  # shows [0 1 2 3 4 5 6 7 8 9]
b = a.reshape((5,2))
b *= 5
print(a)  # shows [ 0  5 10 15 20 25 30 35 40 45]
```
So, what that means is that what we are actually returning here is something that has direct access to the base values, which is not what we want. 

In the `ECM().update_transient_states` method, we get the value of the hysteresis and store it in variable `M`, but then we multiply that by the current, and that changes the `base_values` of the `asoh.h0` attribute.

Reply:
Nice catch!

Reply:
Thank you! Resolving it.